src/platform/packages/shared/kbn-unified-metrics-grid/src/components/metrics_grid.tsx

```diff
@@ -8,143 +8,82 @@
  */
 
 import React, { useMemo } from 'react';
-import type { EuiFlexGridProps, IconType } from '@elastic/eui';
-import {
-  EuiFlexGrid,
-  EuiFlexItem,
-  EuiLoadingChart,
-  EuiFlexGroup,
-  useEuiTheme,
-  euiScrollBarStyles,
-  EuiText,
-  EuiIcon,
-} from '@elastic/eui';
+import type { EuiFlexGridProps } from '@elastic/eui';
+import { EuiFlexGrid, EuiFlexItem, useEuiTheme } from '@elastic/eui';
 import type { MetricField } from '@kbn/metrics-experience-plugin/common/types';
-import { css } from '@emotion/react';
-import { IconChartBarStacked } from '@kbn/chart-icons';
-import { i18n } from '@kbn/i18n';
-import { MetricChart } from './metric_chart';
+import type { ChartSectionProps, UnifiedHistogramInputMessage } from '@kbn/unified-histogram/types';
+import type { Observable } from 'rxjs';
+import { Chart } from './chart';
 
-export type MetricsGridProps = {
-  timeRange: { from?: string; to?: string };
-  loading: boolean;
+export type MetricsGridProps = Pick<
+  ChartSectionProps,
+  'searchSessionId' | 'services' | 'onBrushEnd' | 'onFilter' | 'abortController' | 'requestParams'
+> & {
   filters?: Array<{ field: string; value: string }>;
   dimensions: string[];
   columns: EuiFlexGridProps['columns'];
+  discoverFetch$: Observable<UnifiedHistogramInputMessage>;
 } & (
-  | {
-      pivotOn: 'metric';
-      fields: MetricField[];
-    }
-  | {
-      pivotOn: 'dimension';
-      fields: MetricField;
-    }
-);
+    | {
+        pivotOn: 'metric';
+        fields: MetricField[];
+      }
+    | {
+        pivotOn: 'dimension';
+        fields: MetricField;
+      }
+  );
 
 export const MetricsGrid = ({
   fields,
-  timeRange,
-  loading,
+  searchSessionId,
+  onBrushEnd,
+  onFilter,
   dimensions,
   pivotOn,
-  filters = [],
+  services,
   columns,
+  abortController,
+  requestParams,
+  discoverFetch$,
+  filters = [],
 }: MetricsGridProps) => {
   const euiThemeContext = useEuiTheme();
   const { euiTheme } = euiThemeContext;
```

Comment:
Nit: euiThemeContext is no longer used outside of this


```suggestion
  const { euiTheme }  = useEuiTheme();
```